UserJpaRepository: add missing integration test

Add Integration Tests for [UserJpaRepository](https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepository.java.mustache)

- clean a little bit the [unused variables](https://github.com/jhipster/jhipster-lite/blob/main/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepository.java.mustache#L17-L19)
- add tests for each methods

@pascalgrimaud I think there is some issue whith how `liquibase` scripts are generated.

After generating the project if I try to save `UserEntity` I am getting this error
```
2022-02-02 19:57:06.663  WARN 25252 --- [           main] o.h.engine.jdbc.spi.SqlExceptionHelper   : SQL Error: 1364, SQLState: HY000
2022-02-02 19:57:06.664 ERROR 25252 --- [           main] o.h.engine.jdbc.spi.SqlExceptionHelper   : Field 'id' doesn't have a default value
```
I fixed it by adding `autoIncrement="true"` while creating `id` column of `jhi_user` table in `liquibase`

after that if I try to save `UserEntity` I am getting this error
```
2022-02-02 20:02:43.073  WARN 25620 --- [           main] o.h.engine.jdbc.spi.SqlExceptionHelper   : SQL Error: 1452, SQLState: 23000
2022-02-02 20:02:43.075 ERROR 25620 --- [           main] o.h.engine.jdbc.spi.SqlExceptionHelper   : Cannot add or update a child row: a foreign key constraint fails (`test`.`jhi_user_authority`, CONSTRAINT `fk_authority_name` FOREIGN KEY (`authority_name`) REFERENCES `jhi_authority` (`name`))
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 64.545 s <<< FAILURE! - in tech.jhipster.beer.user.infrastructure.secondary.mysql.UserJpaRepositoryIT
[ERROR] tech.jhipster.beer.user.infrastructure.secondary.mysql.UserJpaRepositoryIT  Time elapsed: 64.545 s  <<< ERROR!
org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [null]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
        at tech.jhipster.beer.user.infrastructure.secondary.mysql.UserJpaRepositoryIT.createUser(UserJpaRepositoryIT.java:30)
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
        at tech.jhipster.beer.user.infrastructure.secondary.mysql.UserJpaRepositoryIT.createUser(UserJpaRepositoryIT.java:30)
Caused by: java.sql.SQLIntegrityConstraintViolationException: Cannot add or update a child row: a foreign key constraint fails (`test`.`jhi_user_authority`, CONSTRAINT `fk_authority_name` FOREIGN KEY (`authority_name`) REFERENCES `jhi_authority` (`name`))
        at tech.jhipster.beer.user.infrastructure.secondary.mysql.UserJpaRepositoryIT.createUser(UserJpaRepositoryIT.java:30)
```

This is the object which I am trying to save
```java
private UserEntity userEntityWithFluentSetters() {
        UserEntity userEntity = new UserEntity()
                .activated(true)
                .activationKey(ACTIVATION_KEY)
                .authorities(Set.of(new AuthorityEntity().name("myAuthority")))
                .email(EMAIL)
                .firstName("myFirstName")
                .lastName("myLastName")
                .imageUrl("myImageUrl")
                .langKey("fr")
                .login(LOGIN)
                .password("C&At!Ev4&Y7&-CvuGWnyqpmEk*3_w28%7%&df$uFQXTjqzEEwM-v34XW!na6")
                .resetDate(Instant.parse("2022-01-24T10:11:12.000Z"))
                .resetKey(RESET_KEY);

        userEntity
                .lastModifiedBy("modifier")
                .lastModifiedDate(Instant.parse("2022-01-23T10:11:12.000Z"))
                .createdBy("creator")
                .createdDate(Instant.parse("2022-01-22T10:11:12.000Z"));

        return userEntity;
    }
```

Got this
```
2022-02-02 20:54:56.903  WARN 12540 --- [           main] o.h.engine.jdbc.spi.SqlExceptionHelper   : SQL Error: 0, SQLState: 23503
2022-02-02 20:54:56.904 ERROR 12540 --- [           main] o.h.engine.jdbc.spi.SqlExceptionHelper   : ERROR: insert or update on table "jhi_user_authority" violates foreign key constraint "fk_authority_name"
  Detail: Key (authority_name)=(myAuthority) is not present in table "jhi_authority".
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 37.194 s <<< FAILURE! - in tech.jhipster.beer.user.infrastructure.secondary.postgresql.UserJpaRepositoryIT
[ERROR] tech.jhipster.beer.user.infrastructure.secondary.postgresql.UserJpaRepositoryIT  Time elapsed: 37.194 s  <<< ERROR!
org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [fk_authority_name]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
        at tech.jhipster.beer.user.infrastructure.secondary.postgresql.UserJpaRepositoryIT.createUser(UserJpaRepositoryIT.java:30)
Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
        at tech.jhipster.beer.user.infrastructure.secondary.postgresql.UserJpaRepositoryIT.createUser(UserJpaRepositoryIT.java:30)
Caused by: org.postgresql.util.PSQLException:
ERROR: insert or update on table "jhi_user_authority" violates foreign key constraint "fk_authority_name"
  Detail: Key (authority_name)=(myAuthority) is not present in table "jhi_authority".
        at tech.jhipster.beer.user.infrastructure.secondary.postgresql.UserJpaRepositoryIT.createUser(UserJpaRepositoryIT.java:30)
```

Seems either something is wrong with this
```java
  @JsonIgnore
  @ManyToMany
  @JoinTable(
    name = "jhi_user_authority",
    joinColumns = { @JoinColumn(name = "user_id", referencedColumnName = "id") },
    inverseJoinColumns = { @JoinColumn(name = "authority_name", referencedColumnName = "name") }
  )
  @BatchSize(size = 20)
  private Set<AuthorityEntity> authorities = new HashSet<>();
```
or may be this
```xml
<addForeignKeyConstraint
      baseColumnNames="authority_name"
      baseTableName="jhi_user_authority"
      constraintName="fk_authority_name"
      referencedColumnNames="name"
      referencedTableName="jhi_authority"
    />
```
debuggin will, will update whatever I'll find

--- a/src/main/java/tech/jhipster/lite/generator/server/springboot/user/domain/SpringBootUserDomainService.java
+++ b/src/main/java/tech/jhipster/lite/generator/server/springboot/user/domain/SpringBootUserDomainService.java
@@ -52,6 +52,7 @@ public class SpringBootUserDomainService implements SpringBootUserService {
     projectRepository.template(project, SOURCE, "UserConstants.java", getSqlDomainJavaPath(packageNamePath));
 
     projectRepository.template(project, SOURCE, "UserEntityTest.java", getSqlJavaTestPath(packageNamePath, sqlDatabase));
+    projectRepository.template(project, SOURCE, "UserJpaRepositoryIT.java", getSqlJavaTestPath(packageNamePath, sqlDatabase));
   }
 
   private void addAuthorityEntity(Project project, DatabaseType sqlDatabase) {

--- a/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepository.java.mustache
+++ b/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepository.java.mustache
@@ -14,10 +14,6 @@ import org.springframework.stereotype.Repository;
  */
 @Repository
 public interface UserJpaRepository extends JpaRepository<UserEntity, Long> {
-  String USERS_BY_LOGIN_CACHE = "usersByLogin";
-
-  String USERS_BY_EMAIL_CACHE = "usersByEmail";
-
   Optional<UserEntity> findOneByActivationKey(String activationKey);
 
   List<UserEntity> findAllByActivatedIsFalseAndActivationKeyIsNotNullAndCreatedDateBefore(Instant dateTime);

--- a/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepositoryIT.java.mustache
+++ b/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepositoryIT.java.mustache
@@ -1,0 +1,120 @@
+package {{packageName}}.user.infrastructure.secondary.{{sqlDatabaseName}};
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.filter;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Set;
+import javax.transaction.Transactional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import {{packageName}}.IntegrationTest;
+
+@IntegrationTest
+@Transactional
+public class UserJpaRepositoryIT {
+
+  private static final String ACTIVATION_KEY = "my_activation_key";
+  private static final String RESET_KEY = "my_reset_key";
+  private static final String LOGIN = "beer";
+  private static final String EMAIL = LOGIN + "@localhost";
+
+  @Autowired
+  AuthorityJpaRepository authorityJpaRepository;
+
+  @Autowired
+  UserJpaRepository userJpaRepository;
+
+  @BeforeEach
+  void createUser() {
+    AuthorityEntity authorityEntity = authorityJpaRepository.save(new AuthorityEntity().name("ROLE_BEER"));
+    UserEntity userEntity = getUserEntity().authorities(Set.of(authorityEntity));
+    userJpaRepository.save(userEntity);
+  }
+
+  @Test
+  void shouldFindOneByActivationKey() {
+    UserEntity userEntity = userJpaRepository.findOneByActivationKey(ACTIVATION_KEY).orElse(null);
+    assertUserEntity(userEntity);
+  }
+
+  @Test
+  @Transactional
+  void shouldFindAllByActivatedIsFalseAndActivationKeyIsNotNullAndCreatedDateBefore() {
+    List<UserEntity> userEntityList = userJpaRepository.findAllByActivatedIsFalseAndActivationKeyIsNotNullAndCreatedDateBefore(
+      Instant.parse("2022-02-22T10:11:12.000Z")
+    );
+    assertThat(userEntityList).isNotEmpty();
+    assertThat(filter(userEntityList).with("login").equalsTo(LOGIN).get()).isNotEmpty();
+  }
+
+  @Test
+  void shouldFindOneByResetKey() {
+    UserEntity userEntity = userJpaRepository.findOneByResetKey(RESET_KEY).orElse(null);
+    assertUserEntity(userEntity);
+  }
+
+  @Test
+  void shouldFindOneByEmailIgnoreCase() {
+    UserEntity userEntity = userJpaRepository.findOneByEmailIgnoreCase(EMAIL).orElse(null);
+    assertUserEntity(userEntity);
+  }
+
+  @Test
+  void shouldFindOneByLogin() {
+    UserEntity userEntity = userJpaRepository.findOneByLogin(LOGIN).orElse(null);
+    assertUserEntity(userEntity);
+  }
+
+  @Test
+  void shouldFindOneWithAuthoritiesByLogin() {
+    UserEntity userEntity = userJpaRepository.findOneWithAuthoritiesByLogin(LOGIN).orElse(null);
+    assertUserEntity(userEntity);
+  }
+
+  @Test
+  void shouldFindOneWithAuthoritiesByEmailIgnoreCase() {
+    UserEntity userEntity = userJpaRepository.findOneWithAuthoritiesByEmailIgnoreCase(EMAIL).orElse(null);
+    assertUserEntity(userEntity);
+  }
+
+  @Test
+  void shouldFindAllByIdNotNullAndActivatedIsTrue() {
+    UserEntity userEntity = getUserEntity().activated(true).login("beer_disabled").email("beer_disabled@localhost");
+    userJpaRepository.save(userEntity);
+    assertThat(
+      filter(userJpaRepository.findAllByIdNotNullAndActivatedIsTrue(Pageable.ofSize(10)).getContent())
+        .with("login")
+        .equalsTo("beer_disabled")
+        .get()
+    )
+      .isNotEmpty();
+  }
+
+  private void assertUserEntity(UserEntity userEntity) {
+    assertThat(userEntity).isNotNull();
+    assertThat(userEntity.getLogin()).isEqualTo(LOGIN);
+    assertThat(userEntity.getAuthorities()).isNotEmpty();
+  }
+
+  private UserEntity getUserEntity() {
+    UserEntity userEntity = new UserEntity()
+      .activated(false)
+      .activationKey(ACTIVATION_KEY)
+      .email(EMAIL)
+      .login(LOGIN)
+      .password("C&At!Ev4&Y7&-CvuGWnyqpmEk*3_w28%7%&df$uFQXTjqzEEwM-v34XW!na6")
+      .resetDate(Instant.parse("2022-01-24T10:11:12.000Z"))
+      .resetKey(RESET_KEY);
+
+    userEntity
+      .lastModifiedBy("modifier")
+      .lastModifiedDate(Instant.parse("2022-01-23T10:11:12.000Z"))
+      .createdBy("creator")
+      .createdDate(Instant.parse("2022-01-22T10:11:12.000Z"));
+    return userEntity;
+  }
+}

--- a/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepositoryIT.java.mustache
+++ b/src/main/resources/generator/server/springboot/user/sqldatabase/UserJpaRepositoryIT.java.mustache
@@ -42,7 +42,6 @@ public class UserJpaRepositoryIT {
   }
 
   @Test
-  @Transactional
   void shouldFindAllByActivatedIsFalseAndActivationKeyIsNotNullAndCreatedDateBefore() {
     List<UserEntity> userEntityList = userJpaRepository.findAllByActivatedIsFalseAndActivationKeyIsNotNullAndCreatedDateBefore(
       Instant.parse("2022-02-22T10:11:12.000Z")

--- a/src/test/java/tech/jhipster/lite/generator/server/springboot/user/application/SpringBootUserAssertFiles.java
+++ b/src/test/java/tech/jhipster/lite/generator/server/springboot/user/application/SpringBootUserAssertFiles.java
@@ -16,6 +16,7 @@ public class SpringBootUserAssertFiles {
     checkJavaDomainFile(project, "UserConstants.java");
 
     checkJavaTestFile(project, "UserEntityTest.java", databaseType);
+    checkJavaTestFile(project, "UserJpaRepositoryIT.java", databaseType);
   }
 
   public static void assertFilesSqlJavaUserAuthority(Project project, DatabaseType databaseType) {

